readinglist/__init__.py

```diff
@@ -97,9 +108,36 @@ def on_new_response(event):
     config.add_subscriber(on_new_response, NewResponse)
 
 
+def end_of_life_tween_factory(handler, registry):
+    """Pyramid tween to handle service end of life."""
+
+    def eos_tween(request):
+        eos_date = registry.settings.get("readinglist.eos")
+        eos_url = registry.settings.get("readinglist.eos_url")
+        if eos_date:
+            eos_date = dateparser.parse(eos_date)
+            alert = {}
+            if eos_url is not None:
+                alert['url'] = eos_url
+
+            if eos_date > datetime.datetime.now():
+                alert['code'] = "soft-eol"
+                response = handler(request)
+            else:
+                response = HTTPServiceDeprecated()
+                alert['code'] = "hard-eol"
+                response.headers = {}
+            response.headers['Alert'] = json.dumps(alert)
```

Comment:
You said in the docs there will be a ``message``, maybe it should be mentionned as optional